### PR TITLE
GEODE-6964: Move geode log4j core classes to geode-log4j

### PR DESCRIPTION
**Pull request by kirklund**
Move all log4j-core code from geode-core to geode-log4j.
Move Geode's log4j2.xml from geode-core to geode-log4j.
Make geode-log4j an implementation dependency for geode-dunit.

Co-authored-by: Mark Hanson <mhanson@pivotal.io>
Co-authored-by: Aaron Lindsey <alindsey@pivotal.io>
Co-authored-by: Dale Emery <demery@pivotal.io>

Note: this PR extracts more log4j dependent code to geode-log4j including some uses of log4j-api.

### CODE REVIEW

File: geode-log4j/src/integrationTest/java/org/apache/geode/logging/log4j/BothLogWriterAppendersIntegrationTest.java [hunk 1]
```diff
@@ -1,21 +1,23 @@
...
  *
- * Unless required by applicable law or agreed to in writing, software distributed under the License
- * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
- * or implied. See the License for the specific language governing permissions and limitations under
- * the License.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-log4j/src/integrationTest/java/org/apache/geode/logging/log4j/BothLogWriterAppendersIntegrationTest.java]: I restored the longer license headers.

File: geode-log4j/src/main/java/org/apache/geode/logging/log4j/message/GemFireParameterizedMessage.java [hunk 2]
```diff
@@ -494,8 +496,8 @@ private static void recursiveDeepToString(final Object o, final StringBuilder st
         }
         str.append('}');
       }
-    } else if (o instanceof Collection && !(o instanceof EntriesSet)) {
-      // GEODE: do NOT use Collection handling if instanceof Geode EntriesSet
+    } else if (o instanceof Collection && !(o instanceof EntriesCollection)) {
+      // GEODE: do NOT use Collection handling if instanceof Geode EntriesCollection
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-log4j/src/main/java/org/apache/geode/logging/log4j/message/GemFireParameterizedMessage.java]: EntriesCollection is a new non-internal interface that EntriesSet implements.

EntriesSet is internal and defined in geode-core, so we don't want geode-log4j to reference it directly.

File: geode-docs/managing/logging/how_logging_works.html.md.erb [hunk 3]
```diff
@@ -21,7 +21,7 @@ limitations under the License.
 
 <%=vars.product_name%> uses [Apache Log4j 2](http://logging.apache.org/log4j/2.x/) API and Core libraries as the basis for its logging system. Log4j 2 API is a popular and powerful front-end logging API used by all the <%=vars.product_name%> classes to generate log statements. Log4j 2 Core is a backend implementation for logging; you can route any of the front-end logging API libraries to log to this backend. <%=vars.product_name%> uses the Core backend to run two custom Log4j 2 Appenders: **AlertAppender** and **LogWriterAppender**.
 
-<%=vars.product_name%> has been tested with Log4j 2.11.
+<%=vars.product_name%> has been tested with Log4j 2.12.
 <%=vars.product_name%> requires the 
 `log4j-api-2.11.0.jar` and `log4j-core-2.11.0.jar`
 JAR files to be in the classpath.
```
[the remark this answers is not in the document]
Comment (kirklund) [reply geode-docs/managing/logging/how_logging_works.html.md.erb]: Yeah, I thought of that too. I think I'd like to do a follow-up PR that sweeps through all of the documentation.

**Comment by kirklund** [comment]
This PR is blocked by some sort of environment issue causing DistributedTestOpenJDK11 to fail and the Gradle daemon to die.

Based on google searches [2] [3], the cause might be either:

1) the distributed test job is out of swap space when the Tomcat/Jetty tests run

2) the /proc/sys/vm/overcommit_memory and /proc/sys/vm/overcommit_ratio may be misconfigured

I don't have sufficient concourse knowledge to attempt to fix or change either of the above two settings in our precheckin environment. Can someone please help me with this?

The failure:
```
OpenJDK 64-Bit Server VM warning: INFO: os::commit_memory(0x00007f74a4ba4000, 65536, 1) failed; error='Not enough space' (errno=12) 
[thread 26510 also had an error] 
```
[1] https://concourse.apachegeode-ci.info/builds/88240
[2] https://bugs.openjdk.java.net/browse/JDK-8216619
[3] https://stackoverflow.com/questions/46801741/jvm-crashes-with-error-cannot-allocate-memory-errno-12

**Comment by kirklund** [comment]
So far, this PR has had one nearly-GREEN precheckin after increasing memory for the DistributedTestOpenJDK11 job. The only failure was RestartOfMemberDistributedTest  (GEODE-7118) which does not reproduce.

File: geode-core/src/main/java/org/apache/geode/distributed/internal/membership/adapter/GMSMembershipManager.java [hunk 4]
```diff
@@ -2585,8 +2584,6 @@ public void forceDisconnect(final String reason) {
       services.setShutdownCause(shutdownCause);
       services.getCancelCriterion().cancel(reason);
 
-      AlertAppender.stopSessionIfRunning();
```
Comment (aaronlindsey) [inline geode-core/src/main/java/org/apache/geode/distributed/internal/membership/adapter/GMSMembershipManager.java]: I thought we didn't want to remove this per the discussion on #3904.

File: geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/GenericAppServerContainer.java [hunk 5]
```diff
@@ -102,6 +105,8 @@ public GenericAppServerContainer(GenericAppServerInstall install, File container
     List<String> command = new ArrayList<>();
     // Path to the modify war script to run
     command.add(modifyWarScript.getAbsolutePath());
+    command.add("-J");
+    command.add("-Xmx2096m");
```
Comment (aaronlindsey) [inline geode-assembly/src/distributedTest/java/org/apache/geode/session/tests/GenericAppServerContainer.java]: What effect does adding these parameters have?

**Comment by kirklund** [comment]
Closing #3914 and replacing it with #4003.

**Pull request closed without merging**